reverie/backend_server/persona/prompt_template/run_gpt_prompt.py

```diff
@@ -359,63 +359,122 @@ def create_prompt_input(persona, task, duration, test_input=None):
         total_expected_min = int(prompt.split("(total duration in minutes")[-1]
                                  .split("):")[0].strip())
 
         # TODO -- now, you need to make sure that this is the same as the sum of
         #         the current action sequence.
-        curr_min_slot = [["dummy", -1], ]  # (task_name, task_index)
-        for count, i in enumerate(cr):
-            i_task = i[0]
+        accumulates_all_activites = [["dummy", -1], ]  # (task_name, task_index)
+
+        for activity_number, i in enumerate(activities_and_durations):
+            i_activity = i[0]
             i_duration = i[1]
 
+            # just normalize to a multiple of 5 e.g., 23%5 = 3 so 23 - 3 = 20
             i_duration -= (i_duration % 5)
+
+            # guess this filters out bad activities and adding one activity
+            # accumulates_all_activites is going to contain N activites where N is the
+            # number of minutes (if gpt was good gpt)
             if i_duration > 0:
                 for j in range(i_duration):
-                    curr_min_slot += [(i_task, count)]
-        curr_min_slot = curr_min_slot[1:]
+                    accumulates_all_activites += [(i_activity, activity_number)]
+
+        # remove... dummy?
+        accumulates_all_activites = accumulates_all_activites[1:]
 
-        if len(curr_min_slot) > total_expected_min:
-            last_task = curr_min_slot[60]
+        # if we gpt was bad, we might have more activities than the required minutes
+        if len(accumulates_all_activites) > total_expected_min:
+            # we get the last task...
+            # is this right? maybe the variable name is just wrong
+            last_task = accumulates_all_activites[60]
+
+            # from minute -1 to minute -6 we are going to set the
+            # activities to last task?
             for i in range(1, 6):
-                curr_min_slot[-1 * i] = last_task
-        elif len(curr_min_slot) < total_expected_min:
-            last_task = curr_min_slot[-1]
-            for i in range(total_expected_min - len(curr_min_slot)):
-                curr_min_slot += [last_task]
-
-        cr_ret = [["dummy", -1], ]
-        for task, task_index in curr_min_slot:
-            if task != cr_ret[-1][0]:
-                cr_ret += [[task, 1]]
-            else:
-                cr_ret[-1][1] += 1
-        cr = cr_ret[1:]
+                accumulates_all_activites[-1 * i] = last_task
 
-        return cr
+        # if we gpt was bad, we might have fewer activities than the required minutes
+        elif len(accumulates_all_activites) < total_expected_min:
+
+            # ok now we get last task for real
+            last_task = accumulates_all_activites[-1]
+
+            # and we are just going to make it longer
+            for i in range(total_expected_min - len(accumulates_all_activites)):
+                accumulates_all_activites += [last_task]
+
+        # manual group by
```

Comment:
I think we could work directly in minutes, removing the need for this list of activities with as many entries as minutes, and the consequent groupby